src/porepy/numerics/elliptic.py

```diff
@@ -71,8 +74,54 @@ def __init__(self, gb, data=None, physics='flow'):
         self._flux_disc = self.flux_disc()
         self._source_disc = self.source_disc()
 
-    def solve(self):
-        self.x = sps.linalg.spsolve(*self.reassemble())
+    def solve(self, max_direct=40000, callback=False, **kwargs):
+        """ Reassemble and solve linear system.
+
+        After the funtion has been called, the attributes lhs and rhs are
+        updated according to the parameter states. Also, the attribute x
+        gives the pressure given the current state.
+
+        TODO: Provide an option to save solver information if multiple
+        systems are to be solved with the same left hand side.
+
+        The function attempts to set up the best linear solver based on the
+        system size. The setup and parameter choices here are still
+        experimental.
+
+        Parameters:
+            max_direct (int): Maximum number of unknowns where a direct solver
+                is applied. If a direct solver can be applied this is usually
+                the most efficient option. However, if the system size is
+                too large compared to available memory, a direct solver becomes
+                extremely slow.
+            callback (boolean, optional): If True iteration information will be
+                output when an iterative solver is applied (system size larger
+                than max_direct)
+
+        Returns:
+            np.array: Pressure state.
+
+        """
+
+        # Discretize
+        tic = time.time()
+        print('Discretize')
+        self.lhs, self.rhs = self.reassemble()
+        print('Done. Elapsed time ' + str(time.time() - tic))
+
+        # Solve
+        tic = time.time()
+        print('Linear solver')
+        ls = LSFactory()
```

Comment:
Is there a reason to initialize LSFactory here or could it be done in the else statement?